share/smack/prelude.py

```diff
@@ -0,0 +1,124 @@
+UNINTERPRETED_UNARY_OP = "function {0}.{1}(i: {1}) returns ({1});\n"
+UNINTERPRETED_BINARY_OP = "function {0}.{1}(i1: {1}, i2: {1}) returns ({1});\n"
+INLINE_BINARY_OP = "function {{:inline}} {0}.{2}(i1: {2}, i2: {2}) returns ({2}) {1}\n"
+BUILTIN_BINARY_OP =  "function {{:builtin \"{0}\"}} {1}.{2}(i1: {2}, i2: {2}) returns ({2}) ;\n"
+BVBUILTIN_UNARY_OP = "function {{:bvbuiltin \"{0}\"}} {1}.{2}(i: {2}) returns ({2}) ;\n"
+BVBUILTIN_BINARY_OP = "function {{:bvbuiltin \"{0}\"}} {1}.{2}(i1: {2}, i2: {2}) returns ({2}) ;\n"
+INLINE_BINARY_COMP = "function {{:inline}} {0}.{2}.bool(i1: {2}, i2: {2}) returns (bool) {1}"
+BVBUILTIN_BINARY_COMP = "function {{:bvbuiltin \"{0}\"}} {1}.{2}.bool(i1: {2}, i2: {2}) returns (bool) ;"
+INLINE_BINARY_PRED = "%s %s\n" % (INLINE_BINARY_COMP, "function {{:inline}} {0}.{2}(i1: {2}, i2: {2}) returns (i1) {{if {0}.{2}.bool(i1,i2) then 1 else 0}}")
+INLINE_BINARY_BV_PRED = "%s %s\n" % (INLINE_BINARY_COMP, "function {{:inline}} {0}.{2}(i1: {2}, i2: {2}) returns (bv1) {{if {0}.{2}.bool(i1,i2) then 1bv1 else 0bv1}}")
+SAFE_LOAD_OP = "function {{:inline}} $load.{1}(M: [ref] {1}, p: ref) returns ({1}) {0}\n"
+SAFE_STORE_OP = "function {{:inline}} $store.{1}(M: [ref] {1}, p: ref, v: {1}) returns ([ref] {1}) {0}\n"
+INLINE_BVBUILTIN_BINARY_SELECT = "function {{:inline}} {0}.{2}(i1: {2}, i2: {2}) returns ({2}) {{if {1}.{2}.bool(i1,i2) then i1 else i2}}\n"
+INLINE_BVBUILTIN_BINARY_PRED = "%s %s\n" % (BVBUILTIN_BINARY_COMP, "function {{:inline}} {1}.{2}(i1: {2}, i2: {2}) returns (bv1) {{if {1}.{2}.bool(i1,i2) then 1bv1 else 0bv1}}")
+INLINE_CONVERSION = "function {{:inline}} {0}.{2}.{3}(i: {2}) returns ({3}) {1}\n"
+TRUNC_OP = "function {{:inline}} $trunc.{0}.{1}(i: {0}) returns ({1}) {{i}}\n"
+TRUNC_BV_OP = "function {{:inline}} $trunc.{0}.{1}(i: {0}) returns ({1}) {{i[{2}:0]}}\n"
+
+sizes = [1, 8, 16, 24, 32, 40, 48, 56, 64, 88, 96, 128]
+
+def declare_each_type(prefix, func_decl, *args):
+  bpl = ""
+  for size in sizes:
+    t = "%s%s" % (prefix, size)
+    tmp = (args + (t,))
+    bpl += func_decl.format(*tmp)
+  return bpl
+
+def declare_each_type_nested(prefix, func_decl, *args):
+  bpl = ""
+  for i in range(len(sizes)):
+    for j in range(0,i):
+      t1 = "%s%s" % (prefix, sizes[i])
+      t2 = "%s%s" % (prefix, sizes[j])
+      if "trunc" not in func_decl:
+        t1, t2 = t2, t1
+      tmp = args + (t1, t2, sizes[j])
+      bpl += func_decl.format(*tmp)
+  return bpl
+
+def generate_prelude(args):
+  bpl = declare_each_type("bv", BVBUILTIN_BINARY_OP, "bvadd", "$add")
```

Comment:
Kind of related to @shaobo-he's comment, I wonder if you could refactor this into loops (over lists of strings representing function names such as `$add`) instead of listing them explicitly line by line.